src/core/options-watcher.cpp

```diff
@@ -0,0 +1,152 @@
+// License: Apache 2.0. See LICENSE file in root directory.
+// Copyright(c) 2023 Intel Corporation. All Rights Reserved.
+
+
+#include <src/core/options-watcher.h>
+#include <proc/synthetic-stream.h>
+
+
+namespace librealsense {
+
+
+options_watcher::options_watcher( std::chrono::milliseconds update_interval )
+    : _update_interval( update_interval )
+{
+}
+
+options_watcher::~options_watcher()
+{
+    try
+    {
+        std::lock_guard< std::mutex > lock( _mutex );
+        _options.clear();
+        stop();
+    }
+    catch( ... )
+    {
+        LOG_DEBUG( "Failures taking lock or stoping thread while destructing options_watcher" );
+    }
+}
+
+void options_watcher::register_option( rs2_option id, std::shared_ptr< option > option )
+{
+    registered_option opt = { option, 0.0f }; //Option actual value will be queried in start if needed.
+
+    {
+        std::lock_guard< std::mutex > lock( _mutex );
+        _options[id] = std::move( opt );
+    }
+
+    if( should_start() )
+        start();
+}
+
+void options_watcher::unregister_option( rs2_option id )
+{
+    {
+        std::lock_guard< std::mutex > lock( _mutex );
+        _options.erase( id );
+    }
+
+    if( should_stop() )
+        stop();
+}
+
+rsutils::subscription options_watcher::subscribe( callback && cb )
+{
+    rsutils::subscription ret = _on_values_changed.subscribe( std::move( cb ) );
+
+    if( should_start() )
+        start();
+
+    return ret;
+}
+
+bool options_watcher::should_start() const
+{
+    return ! should_stop();
+}
+
+bool options_watcher::should_stop() const
+{
+    return _on_values_changed.size() == 0 || _options.size() == 0;
+}
+
+void options_watcher::start()
+{
+    if( ! _updater.joinable() ) // If not already started
+    {
+        if( _should_query_for_first_time )
```

Comment:
I don't see why this flag is needed.
If we start(), then stop(), then start() again, we should refresh the values again.
Also:
We shouldn't do this here, in the main thread, while everyone's waiting.
We should do it in the side thread.

Reply:
Removed flag. Moved update to thread function.